### PR TITLE
Fix misc i18n issues in template pipeline

### PR DESCRIPTION
**Pull request by mmalerba**
Adds additional tests for i18n edge cases and make necessary fixes so they pass. See individual commits for details.

### CODE REVIEW

File: packages/compiler/src/template/pipeline/src/ingest.ts [hunk 1]
```diff
@@ -232,7 +234,12 @@ function ingestTemplate(unit: ViewCompilationUnit, tmpl: t.Template): void {
  * Ingest a literal text node from the AST into the given `ViewCompilation`.
  */
 function ingestContent(unit: ViewCompilationUnit, content: t.Content): void {
-  const op = ir.createProjectionOp(unit.job.allocateXrefId(), content.selector, content.sourceSpan);
+  if (content.i18n !== undefined && !(content.i18n instanceof i18n.TagPlaceholder)) {
```
Comment (KryptonBD) [inline packages/compiler/src/template/pipeline/src/ingest.ts:237]:
```suggestion
  if (content.i18n && !(content.i18n instanceof i18n.TagPlaceholder)) {
```

Comment (dylhunn) [reply packages/compiler/src/template/pipeline/src/ingest.ts:237]: Yep, explicit checks are preferred here

File: packages/compiler/src/template/pipeline/src/phases/resolve_i18n_element_placeholders.ts [hunk 2]
```diff
@@ -86,6 +86,24 @@ function resolvePlaceholdersForView(
           }
         }
         break;
+      case ir.OpKind.Projection:
+        debugger;
```
Comment (dylhunn) [inline packages/compiler/src/template/pipeline/src/phases/resolve_i18n_element_placeholders.ts]: Stray debugger statement

File: packages/compiler/src/template/pipeline/src/phases/extract_i18n_messages.ts [hunk 3]
```diff
@@ -117,132 +117,95 @@ export function extractI18nMessages(job: CompilationJob): void {
...
+      ...value,
+      value: value.value.element,
+      flags: value.flags & ~ir.I18nParamValueFlags.TemplateTag
+    });
+    const templateValue = formatValue({
+      ...value,
+      value: value.value.template,
+      flags: value.flags & ~ir.I18nParamValueFlags.ElementTag
+    });
+    // TODO(mmalerba): This is likely a bug in TemplateDefinitionBuilder, we should need to record
```
Comment (dylhunn) [inline packages/compiler/src/template/pipeline/src/phases/extract_i18n_messages.ts:181]: Nit: "not need", right?

**Review by dylhunn — APPROVED** [review]

File: packages/compiler/src/template/pipeline/src/phases/resolve_i18n_element_placeholders.ts (hunk 2 above)
Comment (mmalerba) [reply packages/compiler/src/template/pipeline/src/phases/resolve_i18n_element_placeholders.ts]: Whoops :)

**Comment by dylhunn** [comment]
This PR was merged into the repository by commit 12dfa9ba133b8a5c82882c5254d200935f589776.

**Review by trinhtuanvy — APPROVED** [review]
